src/main/java/pl/bartoszmech/domain/offer/OfferService.java

```diff
@@ -0,0 +1,49 @@
+package pl.bartoszmech.domain.offer;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+import pl.bartoszmech.domain.offer.dto.OfferApiResponse;
+import pl.bartoszmech.domain.offer.dto.OfferResponse;
+
+import java.util.List;
+import java.util.Objects;
+
+@Log4j2
+@AllArgsConstructor
+@Service
+public class OfferService {
+    OfferFetcher fetcher;
+    OfferRepository repository;
+    public List<OfferResponse> fetchAllOfferAndSaveAllIfNotExists() {
+        log.info("Starting fetch offers...");
```

Comment:
this methos is way to long, please refactor to smaller ones so that the code is readable like a book